RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -167,74 +205,88 @@ void HiEvtPlaneFlatProducer::produce(edm::Event& iEvent, const edm::EventSetup&
       }
     }
   }
-
+  //
+  //Get flattening parameter file.
+  //
   if (hirpWatcher.check(iSetup)) {
     edm::ESHandle<RPFlatParams> flatparmsDB_;
     iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
     LoadEPDB db(flatparmsDB_, flat);
-    if (!db.IsSuccess())
-      return;
-  }
+  }  //rp record change
+
   //
   //Get Centrality
   //
-
-  int bin = iEvent.get(centralityBinToken) / CentBinCompression_;
-
-  if (Noffmin_ >= 0) {
-    int nOff = iEvent.get(centralityToken).Ntracks();
-    if ((nOff < Noffmin_) or (nOff >= Noffmax_)) {
-      return;
-    }
-  }
+  int bin = 0;
+  int cbin = 0;
+  edm::Handle<int> cbin_;
+  iEvent.getByToken(centralityBinToken, cbin_);
+  cbin = *cbin_;
+  bin = cbin / CentBinCompression_;
+  //double centval = cbin*100./(double) nCentBins_;
   //
   //Get Vertex
   //
-  int vs_sell;  // vertex collection size
-  float vzr_sell;
-  auto const& vertices3 = iEvent.get(vertexToken);
-  vs_sell = vertices3.size();
-  if (vs_sell > 0) {
-    vzr_sell = vertices3.begin()->z();
-  } else
-    vzr_sell = -999.9;
+  edm::Handle<reco::VertexCollection> vertices;
+  iEvent.getByToken(vertexToken, vertices);
+
+  //best vertex
+  double bestvz = -999.9, bestvx = -999.9, bestvy = -999.9;
+  const reco::Vertex& vtx = (*vertices)[0];
```

Comment:
here and below see comments in RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc for similar code